proc/proc.go

```diff
@@ -46,6 +46,7 @@ type Process struct {
 	packageMap map[string]string
 
 	allGCache               []*G
+	dwarfLock               sync.Mutex
```

Comment:
That's a good idea, I'm up for that. I don't have a strong opinion on the design of that (Process acquires its own lock vs Debugger acquired a lock when it needs to use Process) so I'll leave that up to you.